web/src/model.rs

```diff
@@ -0,0 +1,188 @@
+                                    </label>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+
+                    {{self.permission_input_view()}}
+
+                    { for self.permissions.iter().map(|permission| self.view_entry(permission.to_string())) }
+
+                    <div class="form-grid__row form-grid__row--small">
+                        <div class="form-item">
+                            <label class="form-label" for="label-and-textarea">{ "Token" }</label>
+                            <div class="form-item__wrapper">
+                                <div class="form-field">
+                                    <textarea
+                                        id="label-and-textarea"
+                                        class="form-field__textarea token-area"
+                                        readonly=true>
+                                        {self.token.clone().map(|jwt| jwt.access_token).unwrap_or("No token".to_string())}
+                                    </textarea>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+
+                    <div class="form-grid__row form-grid__row--small">
+                        <button class="button button--primary button--huge" onclick=self.link.callback(|_| Msg::SetPermissions)>{"Generate token"}</button>
+                    </div>
+                </div>
+            </div>
+        }
+    }
+}
+
+impl Model {
+    fn permission_input_view(&self) -> Html {
+        html! {
+            <div class="form-grid__row form-grid__row--small">
+                <div class="form-grid__row__column form-grid__row__column--span-5">
+                    <div class="form-item">
+                        <label class="form-label" for="permission">{ "Permission" }</label>
+                        <div class="form-item__wrapper">
+                            <div class="form-field">
+                                <label class="form-field__wrapper">
+                                    <input id="form-item-name" class="form-field__text" type="text" placeholder="permission" ref=self.permission_input_ref.clone()/>
+                                </label>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+                <div class="form-grid__row__column display-grid">
+                    <button class="button button--primary button--huge button--icon-only permission-button" type="button" onclick=self.link.batch_callback(|_| { Some(Msg::AddPermission) })>
+                        <div aria-hidden="false" aria-label="Button" class="icon icon--size-s" role="img">
```

Comment:
Errore mio di quando vi ho mandato il codice
```suggestion
                        <div aria-hidden="false" aria-label="Add permission" class="icon icon--size-l" role="img">
```